tools/link.py

```diff
@@ -172,7 +172,7 @@ def setup_environment_settings():
   if any(x for x in settings.ENVIRONMENT if x not in VALID_ENVIRONMENTS):
     exit_with_error(f'Invalid environment specified in "ENVIRONMENT": {settings.ENVIRONMENT}. Should be one of: {",".join(VALID_ENVIRONMENTS)}')
 
-  settings.ENVIRONMENT_MAY_BE_WEB = not settings.ENVIRONMENT or 'web' in settings.ENVIRONMENT
+  settings.ENVIRONMENT_MAY_BE_WEB = not settings.ENVIRONMENT or 'web' in settings.ENVIRONMENT or 'worker' in settings.ENVIRONMENT
```

Comment:
Was this change intentional? It causes web-specific code to be included in the final JS output when building with `-sENVIRONMENT=node` + `-pthread`, see:
https://github.com/emscripten-core/emscripten/blob/7378d2f64c50b036f01d4c817568324501182359/tools/link.py#L165-L169
https://github.com/emscripten-core/emscripten/blob/7378d2f64c50b036f01d4c817568324501182359/emcc.py#L430-L432

Reply:
Indeed, this effectively ends up passing `-sENVIRONMENT=node,worker`. For full context, I noticed this on wasm-vips' test suite while trying to upgrade to Emscripten 4.0.16, where full concurrency isn't leveraged due to:
https://github.com/kleisauke/wasm-vips/blob/v0.0.15/src/vips-library.js#L14-L22

I think it makes sense to just bail out if someone uses `-sENVIRONMENT=worker` (and reverting this part of the code). 👍